granulate_utils/linux/cgroups/cgroup.py

```diff
@@ -28,32 +33,50 @@
 }
 
 
-def get_cgroups(process: Optional[psutil.Process] = None) -> List[Tuple[str, List[str], str]]:
+class ProcCgroupLine:
     """
-    Get the cgroups of a process in [(hier id., controllers, path)] parsed form.
-    If process is None, gets the cgroups of the current process.
+    The format of the line:  hierarchy-ID:controller-list:cgroup-path
+    Example line: 1:cpu:/custom_cgroup
+
+    cgroup-path - the cgroup the process belongs to, relative to the hierarchy mount point e.g. /sys/fs/cgroup
     """
 
-    def parse_line(line: str) -> Tuple[str, List[str], str]:
-        hier_id, controller_list, cgroup_path = line.split(":", maxsplit=2)
-        return hier_id, controller_list.split(","), cgroup_path
+    hier_id: str
+    controllers: List[str]
+    relative_path: str
+
+    def __init__(self, procfs_line: str):
+        hier_id, controller_list, relative_path = procfs_line.split(":", maxsplit=2)
+        self.hier_id = hier_id
+        self.controllers = controller_list.split(",")
+        self.relative_path = relative_path
+
 
+def get_process_cgroups(process: Optional[psutil.Process] = None) -> List[ProcCgroupLine]:
+    """
+    Get the cgroups of a process in [(hier id., controllers, path)] parsed form.
+    If process is None, gets the cgroups of the current process.
+    """
     process = process or psutil.Process()
     text = read_proc_file(process, "cgroup").decode()
-    return [parse_line(line) for line in text.splitlines()]
+    return [ProcCgroupLine(line) for line in text.splitlines()]
+
+
+def is_known_controller(controller: str) -> bool:
+    return controller in CONTROLLERS
 
 
 def find_v1_hierarchies(resolve_host_root_links: bool = True) -> Mapping[str, str]:
```

Comment:
As part of your PR let's do an API cleanup. Why does `resolve_host_root_links` exist? I don't see it ever set in our codebase. This is always true, you never want it "false" afaict